### ISSUE TITLE
Unable to register new event in case of using MySQL as datastore

### ISSUE THREAD

**Issue by dominhhai** [pipe-cd/pipecd#3244]
**What happened**:
Can not register event because of db insertion error.
```
failed to create entity {"id": "08106005-ade6-4f0a-ad82-6381346bc8f4", "kind": "Event", "error": "Error 1048: Column 'Status' cannot be null"}
github.com/pipe-cd/pipecd/pkg/datastore/mysql.(*MySQL).Create
        pkg/datastore/mysql/mysql.go:165
github.com/pipe-cd/pipecd/pkg/datastore.(*eventStore).Add
        pkg/datastore/eventstore.go:69
github.com/pipe-cd/pipecd/pkg/app/server/grpcapi.(*API).RegisterEvent
        pkg/app/server/grpcapi/api.go:420
github.com/pipe-cd/pipecd/pkg/app/server/service/apiservice._APIService_RegisterEvent_Handler.func1
        bazel-out/k8-fastbuild/bin/pkg/app/server/service/apiservice/apiservice_go_proto_/github.com/pipe-cd/pipecd/pkg/app/server/service/apiservice/service.pb.go:2507
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1
        external/com_github_grpc_ecosystem_go_grpc_prometheus/server_metrics.go:107
github.com/pipe-cd/pipecd/pkg/rpc.ChainUnaryServerInterceptors.func1.1.1
        pkg/rpc/chain_interceptor.go:30
github.com/pipe-cd/pipecd/pkg/rpc.RequestValidationUnaryServerInterceptor.func1
        pkg/rpc/request_validation_interceptor.go:41
github.com/pipe-cd/pipecd/pkg/rpc.ChainUnaryServerInterceptors.func1.1.1
        pkg/rpc/chain_interceptor.go:30
github.com/pipe-cd/pipecd/pkg/rpc/rpcauth.APIKeyUnaryServerInterceptor.func1
        pkg/rpc/rpcauth/interceptor.go:167
github.com/pipe-cd/pipecd/pkg/rpc.ChainUnaryServerInterceptors.func1.1.1
        pkg/rpc/chain_interceptor.go:30
github.com/pipe-cd/pipecd/pkg/rpc.LogUnaryServerInterceptor.func1
        pkg/rpc/log_interceptor.go:32
github.com/pipe-cd/pipecd/pkg/rpc.ChainUnaryServerInterceptors.func1.1.1
        pkg/rpc/chain_interceptor.go:30
github.com/pipe-cd/pipecd/pkg/rpc.ChainUnaryServerInterceptors.func1
        pkg/rpc/chain_interceptor.go:37
github.com/pipe-cd/pipecd/pkg/app/server/service/apiservice._APIService_RegisterEvent_Handler
        bazel-out/k8-fastbuild/bin/pkg/app/server/service/apiservice/apiservice_go_proto_/github.com/pipe-cd/pipecd/pkg/app/server/service/apiservice/service.pb.go:2509
google.golang.org/grpc.(*Server).processUnaryRPC
        external/org_golang_google_grpc/server.go:1180
google.golang.org/grpc.(*Server).handleStream
        external/org_golang_google_grpc/server.go:1503
google.golang.org/grpc.(*Server).serveStreams.func1.2
        external/org_golang_google_grpc/server.go:843
failed to handle an unary gRPC request: /grpc.service.apiservice.APIService/RegisterEvent       {"error": "rpc error: code = Internal desc = Failed to add event 08106005-ade6-4f0a-ad82-6381346bc8f4", "duration": 0.071947276}
github.com/pipe-cd/pipecd/pkg/rpc.LogUnaryServerInterceptor.func1
        pkg/rpc/log_interceptor.go:37
github.com/pipe-cd/pipecd/pkg/rpc.ChainUnaryServerInterceptors.func1.1.1
        pkg/rpc/chain_interceptor.go:30
github.com/pipe-cd/pipecd/pkg/rpc.ChainUnaryServerInterceptors.func1
        pkg/rpc/chain_interceptor.go:37
github.com/pipe-cd/pipecd/pkg/app/server/service/apiservice._APIService_RegisterEvent_Handler
        bazel-out/k8-fastbuild/bin/pkg/app/server/service/apiservice/apiservice_go_proto_/github.com/pipe-cd/pipecd/pkg/app/server/service/apiservice/service.pb.go:2509
google.golang.org/grpc.(*Server).processUnaryRPC
        external/org_golang_google_grpc/server.go:1180
google.golang.org/grpc.(*Server).handleStream
        external/org_golang_google_grpc/server.go:1503
google.golang.org/grpc.(*Server).serveStreams.func1.2
        external/org_golang_google_grpc/server.go:843
handled an unary gRPC request: /grpc.service.pipedservice.PipedService/ListUnhandledCommands    {"code": "OK", "duration": 0.080193804}
handled an unary gRPC request: /grpc.service.pipedservice.PipedService/ListNotCompletedDeployments      {"code": "OK", "duration": 0.081122692}
```

**What you expected to happen**:
Handle DB insertion

**How to reproduce it**:
Run `$ pipectl event register` command.

**Environment**:
- `piped` version: 0.25.1
- `control-plane` version: 0.25.1
- Others:

**Comment by khanhtc1202** [comment]
Thanks for reporting 🙏 This issue caused by the current MySQL datastore has one step to encode entity object before insertion, and Event entity which will be registered to the control-plane has value for field `Status` is 0, which will be omitted follow the above steps. We will release a patch to fix this asap, but in the meantime, until it's released, you could use the following SQL command to fix it from the database side

```sql
> ALTER TABLE Event MODIFY COLUMN Status INT GENERATED ALWAYS AS (IFNULL(data->>"$.status", 0)) VIRTUAL NOT NULL;
``` 

Thank you so much for your help 🙌

**Comment by khanhtc1202** [comment]
Fix for this had released (version v0.26.0)
